From d474b3c2f6d3bc15ba17d80c5c3503cf3dfee21d Mon Sep 17 00:00:00 2001
From: Daniel Zhang <danielzhang130@gmail.com>
Date: Mon, 2 Sep 2024 15:24:53 -0400
Subject: [PATCH] Fix p2p hidl crash

- WifiP2PGroup.setClientInterfaceMacAddress will crash because hidl
  callback impl does not provide mac address

Test: manual. Try receiving file with nearby share
Change-Id: Iecbb6a0f96d93def74f02a3144542c075868a4bd
Signed-off-by: Daniel Zhang <danielzhang130@gmail.com>
---
 .../server/wifi/p2p/SupplicantP2pIfaceCallbackHidlImpl.java    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/service/java/com/android/server/wifi/p2p/SupplicantP2pIfaceCallbackHidlImpl.java b/service/java/com/android/server/wifi/p2p/SupplicantP2pIfaceCallbackHidlImpl.java
index e1d9499..657d34d 100644
--- a/service/java/com/android/server/wifi/p2p/SupplicantP2pIfaceCallbackHidlImpl.java
+++ b/service/java/com/android/server/wifi/p2p/SupplicantP2pIfaceCallbackHidlImpl.java
@@ -19,6 +19,7 @@ package com.android.server.wifi.p2p;
 import android.annotation.NonNull;
 import android.hardware.wifi.supplicant.V1_0.ISupplicantP2pIfaceCallback;
 import android.hardware.wifi.supplicant.V1_0.WpsConfigMethods;
+import android.net.MacAddress;
 import android.net.wifi.WpsInfo;
 import android.net.wifi.p2p.WifiP2pConfig;
 import android.net.wifi.p2p.WifiP2pDevice;
@@ -501,10 +502,12 @@ public class SupplicantP2pIfaceCallbackHidlImpl extends ISupplicantP2pIfaceCallb
         }
         try {
             device.deviceAddress = NativeUtil.macAddressFromByteArray(deviceAddressBytes);
+            device.setInterfaceMacAddress(MacAddress.fromBytes(srcAddress));
         } catch (Exception e) {
             Log.e(TAG, "Could not decode MAC address", e);
             return null;
         }
+        device.setIpAddress(null);
         return device;
     }
 
-- 
2.34.1

